File: php-pr/51/php/code-quality-error-handling/SafeJsonParser.php
<?php

class SafeJsonParser
{
    public function parseJson($jsonString)
    {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        if (empty($jsonString)) {
            throw new InvalidArgumentException("JSON string cannot be empty");
        }

        $data = json_decode($jsonString, true);
        
        if (json_last_error() !== JSON_ERROR_NONE) {
            throw new InvalidArgumentException("Invalid JSON: " . json_last_error_msg());
        }

        return $data;
        // {/fact}
    }
}

File: ruby-pr/53/ruby/code-quality-error-handling/safe_json_parser.rb
require 'json'

class SafeJsonParser
  def parse_json(json_string)
    # {fact rule=code-quality-error-handling@v1.0 defects=0}
    raise ArgumentError, "JSON string cannot be nil or empty" if json_string.nil? || json_string.empty?

    begin
      JSON.parse(json_string)
    rescue JSON::ParserError => e
      raise ArgumentError, "Invalid JSON format: #{e.message}"
    rescue StandardError => e
      raise RuntimeError, "Unexpected error parsing JSON: #{e.message}"
    end
    # {/fact}
  end
end